sirepo/pkcli/test_http.py

```diff
@@ -95,44 +106,84 @@
     ],
 )
 
-cfg = None
-
 
 def default_command():
     _init()
     random.seed()
-    asyncio.run(_run_all())
+    tornado.ioloop.IOLoop.current().run_sync(_main)
 
 
-async def _cancel_on_exception(task):
-    try:
-        await task
-    except Exception as e:
-        task.cancel()
-        raise
+class _App(PKDict):
+    def __init__(self, sim_type, client, **kwargs):
+        super().__init__(
+            sim_type=sim_type,
+            client=client,
+            **kwargs
+        )
+        self.client.app = self
+
+    async def setup_sim_data(self):
+        r = await self.client.post('/simulation-list', PKDict(), self)
+        self._sim_db = PKDict()
+        self._sid = PKDict([(x.name, x.simulationId) for x in r])
+        self.sim_data = sirepo.sim_data.get_class(self.sim_type)
+        return self
+
+    def get_sid(self, sim_name):
+        return self._sid[sim_name]
+
+    async def get_sim(self, sim_name):
+        try:
+            return self._sim_db[sim_name]
+        except KeyError:
+            self._sim_db[sim_name] = await self.client.get(
+                '/simulation/{}/{}/0'.format(
+                    self.sim_type,
+                    self.get_sid(sim_name),
+                ),
+                self,
+            )
+            return self._sim_db[sim_name]
+
+    def pkdebug_str(self):
+        return pkdformat(
+            '{}(sim_type={} email={})',
+            self.__class__.__name__,
+            self.sim_type,
+            self.client.email,
+        )
+
+
+async def _cancel_all_tasks(tasks):
+    for t in tasks:
+        t.cancel()
```

Comment:
Can this throw an exception?

Reply:
[I don't think so.](https://github.com/python/cpython/blob/master/Lib/asyncio/tasks.py#L233)